src/code.c

```diff
@@ -853,48 +833,49 @@ void CodeFuncExprBegin (
     assert( stat1 == OFFSET_FIRST_STAT );
 }
 
-void CodeFuncExprEnd(UInt nr)
+Expr CodeFuncExprEnd(UInt nr, UInt useStack)
 {
-    Expr                expr;           /* function expression, result     */
-    Stat                stat1;          /* single statement of body        */
-    Obj                 fexp;           /* function expression bag         */
-    UInt                len;            /* length of func. expr. list      */
-    UInt                i;              /* loop variable                   */
+    Expr expr;  /* function expression, result     */
+    Stat stat1; /* single statement of body        */
+    Obj  fexp;  /* function expression bag         */
+    UInt len;   /* length of func. expr. list      */
+    UInt i;     /* loop variable                   */
 
     /* get the function expression                                         */
     fexp = CURR_FUNC();
-    
-    /* get the body of the function                                        */
-    /* push an additional return-void-statement if necessary              */
-    /* the function interpreters depend on each function ``returning''     */
-    if ( nr == 0 ) {
-        CodeReturnVoid();
-        nr++;
-    }
-    else {
-        stat1 = PopStat();
-        PushStat( stat1 );
-        if ( TNUM_STAT(stat1) != T_RETURN_VOID
-          && TNUM_STAT(stat1) != T_RETURN_OBJ )
-        {
-            CodeReturnVoidWhichIsNotProfiled();
+
+    if (!useStack) {
+        /* get the body of the function */
+        /* push an additional return-void-statement if necessary */
+        /* the function interpreters depend on each function ``returning'' */
+        if (nr == 0) {
+            CodeReturnVoid();
             nr++;
         }
-    }
+        else {
+            stat1 = PopStat();
+            PushStat(stat1);
+            if (TNUM_STAT(stat1) != T_RETURN_VOID &&
+                TNUM_STAT(stat1) != T_RETURN_OBJ) {
+                CodeReturnVoidWhichIsNotProfiled();
+                nr++;
+            }
+        }
 
-    /* if the body is a long sequence, pack the other statements           */
-    if ( 7 < nr ) {
-        stat1 = PopSeqStat( nr-6 );
-        PushStat( stat1 );
-        nr = 7;
+        /* if the body is a long sequence, pack the other statements */
+        if (7 < nr) {
+            stat1 = PopSeqStat(nr - 6);
+            PushStat(stat1);
+            nr = 7;
+        }
     }
 
     /* stuff the first statements into the first statement sequence       */
     /* Making sure to preserve the line number and file name              */
     STAT_HEADER(OFFSET_FIRST_STAT)->line = LINE_STAT(OFFSET_FIRST_STAT);
-    STAT_HEADER(OFFSET_FIRST_STAT)->size = nr*sizeof(Stat);
-    STAT_HEADER(OFFSET_FIRST_STAT)->type = T_SEQ_STAT+nr-1;
-    for ( i = 1; i <= nr; i++ ) {
+    STAT_HEADER(OFFSET_FIRST_STAT)->size = nr * sizeof(Stat);
```

Comment:
Revert gratuitous reformatting?
```suggestion
    STAT_HEADER(OFFSET_FIRST_STAT)->size = nr*sizeof(Stat);
```